src/kpro_connection.erl

```diff
@@ -424,6 +437,13 @@ handle_msg({tcp_error, Sock, Reason}, #state{sock = Sock}, _) ->
   exit({tcp_error, Reason});
 handle_msg({ssl_error, Sock, Reason}, #state{sock = Sock}, _) ->
   exit({ssl_error, Reason});
+handle_msg({From, {send, Request}},
+           #state{reauthenticating = true, requests = Requests} = State, Debug) ->
+  MaxAge = kpro_sent_reqs:scan_for_max_age(Requests),
```

Comment:
This is quite likely unnecessary, and just blocking while reauthenticating should be fine per the KIP https://cwiki.apache.org/confluence/display/KAFKA/KIP-368%3A+Allow+SASL+Connections+to+Periodically+Re-Authenticate